autonomi/src/client/fs.rs

```diff
@@ -8,15 +8,19 @@
 
 use crate::client::archive::Metadata;
 use crate::client::data::CostError;
+use crate::client::utils::process_tasks_with_max_concurrency;
 use crate::client::Client;
 use bytes::Bytes;
 use sn_evm::EvmWallet;
 use sn_networking::target_arch::{Duration, SystemTime};
 use std::path::PathBuf;
+use tokio::task::JoinError;
 
 use super::archive::{Archive, ArchiveAddr};
 use super::data::{DataAddr, GetError, PutError};
 
+pub(crate) const FILE_UPLOAD_BATCH_SIZE: usize = 128;
```

Comment:
What's the rationale behind 128? Do we need batching at all?

Reply:
I tried with unlimited and my memory useage spiked to the moon. 
So it seems you were right about batching, and we probably need some. Local testing revealed that 128 was a sweet spot (at least on my machine), not too high for mem, not too slow.

Reply:
I do worry a little about the relative performance (e.g. PC vs Raspberry Pi). Can we relate this value to CPU core count?

https://doc.rust-lang.org/std/thread/fn.available_parallelism.html

Reply:
Oooh excellent! Yes!